### PR TITLE
feat(connector management): add details overlay

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

Updated main page UI and added details overlay and functionality

## Why

To enhance connector management and to align with Figma design

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/577

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/types/Constants.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export enum PAGES {
   PARTNER_NETWORK = 'partnernetwork',
   USER_MANAGEMENT = 'usermanagement',
   TECHNICAL_SETUP = 'technicalsetup',
+  CONNECTOR_MANAGEMENT = 'connectormanagement',
```
Comment (jjeroch) [inline src/types/Constants.ts:49]: @lavanya-bmw in line 207 we had the config
 TECHNICAL_SETUP_VIEW = 'view_technical_setup',

This was handling the page access for this page via permission. Now by adding line 49 and changing the page name in the config file, will this still work?

File: src/types/Config.tsx [hunk 2]
```diff
@@ -402,10 +402,15 @@ export const ALL_PAGES: IPage[] = [
...
-    name: PAGES.TECHNICAL_SETUP,
+    name: PAGES.CONNECTOR_MANAGEMENT,
     role: ROLES.TECHNICAL_SETUP_VIEW,
     element: <EdcConnector />,
   },
+  {
+    name: PAGES.TECHNICAL_SETUP,
+    role: ROLES.TECHNICAL_SETUP_VIEW,
+    element: <Redirect path={PAGES.CONNECTOR_MANAGEMENT} />,
+  },
```
Comment (jjeroch) [inline src/types/Config.tsx:413]: Minor comment - nothing which blocks the PR:
I suggest an inline comment to be added. This should get removed again with 24.12. Reason, we expect that after that all users which are using bookmarks have switched to the new page. @oyo what do you think?

**Review by jjeroch — CHANGES_REQUESTED** [review]

File: src/types/Constants.ts (hunk 1 above)
Comment (lavanya-bmw) [reply src/types/Constants.ts:49]: @jjeroch I am using the same role and only changed the page name, it still works
{
    name: PAGES.CONNECTOR_MANAGEMENT,
    role: ROLES.TECHNICAL_SETUP_VIEW,
    element: <EdcConnector />,
  },

File: src/types/Config.tsx (hunk 2 above)
Comment (lavanya-bmw) [reply src/types/Config.tsx:414]: Added

File: src/components/pages/EdcConnector/edcConnectorTableColumns.tsx [hunk 3]
```diff
@@ -116,10 +106,35 @@ export const ConnectorTableColumns = (
...
+      disableColumnMenu: true,
+      align: 'center',
+      headerAlign: 'center',
+      renderCell: () => {
+        return (
+          <IconButton color="secondary">
+            <ArrowForwardIcon />
+          </IconButton>
+        )
+      },
```
Comment (oyo) [inline src/components/pages/EdcConnector/edcConnectorTableColumns.tsx:132]: guess we can write that more compact (formatting ignored here):

`renderCell: () => <IconButton color="secondary"><ArrowForwardIcon /></IconButton>`

File: src/types/Config.tsx (hunk 2 above)
Comment (oyo) [reply src/types/Config.tsx:414]: okay for me

**Review by oyo — CHANGES_REQUESTED** [review]
one small change

**Review by jjeroch — APPROVED** [review]

**Review by oyo — APPROVED** [review]

**Comment by evegufy** [comment]
@lavanya-bmw could you please solve the conflicts?

**Review by evegufy — APPROVED** [review]

**Pull request merged**